pkg/core/application_store.go

```diff
@@ -0,0 +1,54 @@
+package core
+
+import (
+	"context"
+
+	"github.com/fuseml/fuseml-core/gen/application"
+)
+
+// ApplicationStore describes in memory store for applications
+type ApplicationStore struct {
+	items map[string]*application.Application
```

Comment:
I recommend that you use a domain data type for applications instead of depending on the one generated out of goa for REST API purposes. It may seem unnecessary now, but it will be very useful when the codebase starts growing and changes to the REST API can be made without affecting the core functionality. I've done something similar for codesets here, you can use that as a reference: https://github.com/fuseml/fuseml-core/pull/44 .

Reply:
OK, I'll update it.